pkg/kubectl/cmd/util/printing.go

```diff
@@ -28,12 +29,12 @@ import (
 
 // AddPrinterFlags adds printing related flags to a command (e.g. output format, no headers, template path)
 func AddPrinterFlags(cmd *cobra.Command) {
-	cmd.Flags().StringP("output", "o", "", "Output format. One of: json|yaml|template|templatefile|wide|jsonpath.")
+	cmd.Flags().StringP("output", "o", "", "Output format. One of: json|yaml|wide|gotemplate=...|gotemplate-file=...|jsonpath=...|jsonpath-file=... See golang template [http://golang.org/pkg/text/template/#pkg-overview] and jsonpath template [http://releases.k8s.io/HEAD/docs/user-guide/jsonpath.md].")
```

Comment:
@daizuozhuo, in that case the command(s) should look like `-o gotemplate --template=...`.

`-o gotemplate=...` doesn't work because you are trying to pass 2 values to the same flag (`--output`). You have to pass the `gotemplate` value to the output flag, and then the template string to the `--template` flag.

It passes shippable because there are no shippable tests that exercise `kubectl * -o template`, but we do use it on cluster create.